### ISSUE TITLE
PyCode analyzer breaks on "partial(lambda p: p('xx'), print)" ... causes autodoc_member_order = "bysource" not to work

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1152]
Hey,

When I use latest sphinx release v1.2b1 (but same problem in 1.1.3) to generate documentation for following:

```
#!python

from functools import partial

def b():
    "b docstring"
    pass

partial(lambda p: p('xx'), print)

def a():
    "a docstring"
    pass

```

member_order = "bysource" is ignored.

Doing a bit of debugging, it turned out that pycode analyzer raises an exception ParseError("bad input: type=1, value='print', context=('', (83, 20))",)

I have added to line 159 in pycode/pgen2/parse.py (just before the exception would be thrown)

```
                    if value == "print":
                        return
```

This is indeed just a quick-fix to solve the problem for me and a more general fix should go in.

Thanks,
Lukas

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1152
- Originally reported by: Lukas Vacek
- Originally created at: 2013-04-24T18:07:41.054

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-12 18:43:34+00:00_

Closes #1152: Fix pycode parsing errors of Python 3 code by including two grammar
versions for Python 2 and 3, and loading the appropriate version for the
running Python version.

→ <<cset 42a8138c30ca2415c068f6dc19f9dbffddca0cac>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-12 18:44:09+00:00_

Yeah, the same grammar for 2 and 3 was hairy. Better to fix it once and for all.
